service/worker/archiver/activities.go

```diff
@@ -173,6 +174,9 @@ func uploadHistoryActivity(ctx context.Context, request ArchiveRequest) (err err
 			logger.Error(uploadErrorMsg, tag.ArchivalUploadFailReason(reason), tag.ArchivalBlobKey(key.String()))
 			return cadence.NewCustomError(errConstructBlob, err.Error())
 		}
+		currBlobSize := int64(len(blob.Body))
+		scope.RecordTimer(metrics.ArchiverBlobSize, time.Duration(currBlobSize))
```

Comment:
Probably it will be useful to dual emit these as domain specific metric.

Reply:
I would go further and say you should start emitting all archival metrics as domain tagged if its easy to do that..

Reply:
I am already emitting all metrics as domain specific metrics.